anesthetic/samples.py

```diff
@@ -388,14 +393,22 @@ def plot_2d(self, axes=None, *args, **kwargs):
                     if x in self and y in self and lkwargs['kind'] is not None:
                         xlabel = self.get_label(x)
                         ylabel = self.get_label(y)
+                        if ~np.isfinite(self[x]).all():
+                            warnings.warn(f"column {x} has inf values.")
                         if x == y:
-                            self[x].plot(ax=ax.twin, xlabel=xlabel,
-                                         *args, **lkwargs)
+                            selfx = self[x].replace([-np.inf, np.inf], np.nan)
+                            selfx.plot(ax=ax.twin, xlabel=xlabel,
+                                       *args, **lkwargs)
                             ax.set_xlabel(xlabel)
                             ax.set_ylabel(ylabel)
                         else:
-                            self.plot(x, y, ax=ax, xlabel=xlabel,
-                                      ylabel=ylabel, *args, **lkwargs)
+                            if ~np.isfinite(self[y]).all():
+                                warnings.warn(f"column {y} has inf values.")
```

Comment:
see other comment

Reply:
done in https://github.com/handley-lab/anesthetic/pull/324/commits/a498a0b1304b26b627b3ce2732200c7258b80a6c